examples/wunderlist.rs

```diff
@@ -0,0 +1,258 @@
+//!
+//! This example showcases the Wunderlist OAuth2 process for requesting access to the user's todo lists.
+//! Wunderlist does not implement the correct token response, so this serves as an example of how to
+//! implement a custom client.
+//!
+//! Before running it, you'll need to create your own wunderlist app.
+//!
+//! In order to run the example call:
+//!
+//! ```sh
+//! WUNDER_CLIENT_ID=xxx WUNDER_CLIENT_SECRET=yyy cargo run --example wunderlist
+//! ```
+//!
+//! ...and follow the instructions.
+//!
+
+extern crate base64;
+extern crate oauth2;
+extern crate rand;
+extern crate url;
+extern crate serde;
+
+use oauth2::TokenType;
+use oauth2::basic::{ BasicErrorResponse, BasicTokenType };
+use oauth2::curl::http_client;
+use oauth2::helpers;
+use oauth2::{
+    AccessToken,
+    AuthorizationCode,
+    AuthUrl,
+    Client,
+    ClientId,
+    ClientSecret,
+    CsrfToken,
+    EmptyExtraTokenFields,
+    ExtraTokenFields,
+    RedirectUrl,
+    RefreshToken,
+    Scope,
+    TokenResponse,
+    TokenUrl,
+};
+
+use std::time::Duration;
+use serde::{Serialize, Deserialize};
+
+use std::env;
+use std::io::{BufRead, BufReader, Write};
+use std::net::TcpListener;
+use url::Url;
+
+type SpecialTokenResponse = NonStandardTokenResponse<EmptyExtraTokenFields>;
+type SpecialClient = Client<BasicErrorResponse, SpecialTokenResponse, BasicTokenType>;
+
+fn default_token_type() -> Option<BasicTokenType>{
+    Some(BasicTokenType::Bearer)
+}
+
+///
+/// Non Standard OAuth2 token response.
+///
+/// This struct includes the fields defined in
+/// [Section 5.1 of RFC 6749](https://tools.ietf.org/html/rfc6749#section-5.1), as well as
+/// extensions defined by the `EF` type parameter.
+/// In this particular example token_type is optional to showcase how to deal with a non
+/// compliant provider.
+///
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub struct NonStandardTokenResponse<EF: ExtraTokenFields> {
+    access_token: AccessToken,
+    #[serde(default = "default_token_type")]
+    token_type: Option<BasicTokenType>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    expires_in: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    refresh_token: Option<RefreshToken>,
+    #[serde(rename = "scope")]
+    #[serde(deserialize_with = "helpers::deserialize_space_delimited_vec")]
+    #[serde(serialize_with = "helpers::serialize_space_delimited_vec")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    scopes: Option<Vec<Scope>>,
+
+    #[serde(bound = "EF: ExtraTokenFields")]
+    #[serde(flatten)]
+    extra_fields: EF,
+}
+
+impl<EF> NonStandardTokenResponse<EF>
+where
+    EF: ExtraTokenFields,
+{
+    ///
+    /// Get token
+    ///
+    pub fn get_access_token(self) -> AccessToken {
```

Comment:
it doesn't look like we need this function, since users can just call `.access_token().clone()` if they need an owned version, right?

Reply:
Done, thanks for pointing it out.